airbyte-cdk/python/unit_tests/sources/file_based/test_file_based_stream_reader.py

```diff
@@ -0,0 +1,71 @@
+    "a/b/c/d", "a/b/c/d.csv", "a/b/c/d.csv.gz", "a/b/c/d.jsonl"
+]
+FILES = make_remote_files(FILEPATHS)
+
+cases = [
+    ([], set(FILEPATHS), set()),
+    (["**"], set(FILEPATHS), set()),
+    (["**/*.csv"], {"a.csv", "a/b.csv", "a/c.csv", "a/b/c.csv", "a/c/c.csv", "a/b/c/d.csv"}, set()),
+    (["**/*.csv*"], {"a.csv", "a.csv.gz", "a/b.csv", "a/b.csv.gz", "a/c.csv", "a/c.csv.gz", "a/b/c.csv", "a/b/c.csv.gz", "a/c/c.csv", "a/c/c.csv.gz", "a/b/c/d.csv", "a/b/c/d.csv.gz"}, set()),
+    (["*"], {"a", "a.csv", "a.csv.gz", "a.jsonl"}, set()),
+    (["*.csv"], {"a.csv"}, set()),
+    (["*.csv*"], {"a.csv", "a.csv.gz"}, set()),
+
+    (["*/*"], {"a/b", "a/b.csv", "a/b.csv.gz", "a/b.jsonl", "a/c", "a/c.csv", "a/c.csv.gz", "a/c.jsonl"}, set()),
+    (["*/*.csv"], {"a/b.csv", "a/c.csv"}, set()),
+    (["*/*.csv*"], {"a/b.csv", "a/b.csv.gz", "a/c.csv", "a/c.csv.gz"}, set()),
+    (["*/**"], {"a/b", "a/b.csv", "a/b.csv.gz", "a/b.jsonl", "a/c", "a/c.csv", "a/c.csv.gz", "a/c.jsonl", "a/b/c", "a/b/c.csv", "a/b/c.csv.gz", "a/b/c.jsonl", "a/c/c", "a/c/c.csv", "a/c/c.csv.gz", "a/c/c.jsonl", "a/b/c/d", "a/b/c/d.csv", "a/b/c/d.csv.gz", "a/b/c/d.jsonl"}, set()),
+    (["a/*"], {"a/b", "a/b.csv", "a/b.csv.gz", "a/b.jsonl", "a/c", "a/c.csv", "a/c.csv.gz", "a/c.jsonl"}, {"a"}),
+    (["a/*.csv"], {"a/b.csv", "a/c.csv"}, {"a"}),
+    (["a/*.csv*"], {"a/b.csv", "a/b.csv.gz", "a/c.csv", "a/c.csv.gz"}, {"a"}),
+    (["a/b/*"], {"a/b/c", "a/b/c.csv", "a/b/c.csv.gz", "a/b/c.jsonl"}, {"a/b"}),
+    (["a/b/*.csv"], {"a/b/c.csv"}, {"a/b"}),
+    (["a/b/*.csv*"], {"a/b/c.csv", "a/b/c.csv.gz"}, {"a/b"}),
+    (["a/*/*"], {"a/b/c", "a/b/c.csv", "a/b/c.csv.gz", "a/b/c.jsonl", "a/c/c", "a/c/c.csv", "a/c/c.csv.gz", "a/c/c.jsonl"}, {"a"}),
+    (["a/*/*.csv"], {"a/b/c.csv", "a/c/c.csv"}, {"a"}),
+    (["a/*/*.csv*"], {"a/b/c.csv", "a/b/c.csv.gz", "a/c/c.csv", "a/c/c.csv.gz"}, {"a"}),
+    (["a/**/*"], {"a/b", "a/b.csv", "a/b.csv.gz", "a/b.jsonl", "a/c", "a/c.csv", "a/c.csv.gz", "a/c.jsonl", "a/b/c", "a/b/c.csv", "a/b/c.csv.gz", "a/b/c.jsonl", "a/c/c", "a/c/c.csv", "a/c/c.csv.gz", "a/c/c.jsonl", "a/b/c/d", "a/b/c/d.csv", "a/b/c/d.csv.gz", "a/b/c/d.jsonl"}, {"a"}),
+    (["a/**/*.csv"], {"a/b.csv", "a/c.csv", "a/b/c.csv", "a/c/c.csv", "a/b/c/d.csv"}, {"a"}),
+    (["a/**/*.csv*"], {"a/b.csv", "a/b.csv.gz", "a/c.csv", "a/c.csv.gz", "a/b/c.csv", "a/b/c.csv.gz", "a/c/c.csv", "a/c/c.csv.gz", "a/b/c/d.csv", "a/b/c/d.csv.gz"}, {"a"}),
+    (["**/*.csv", "**/*.gz"], {"a.csv", "a.csv.gz", "a/b.csv", "a/b.csv.gz", "a/c.csv", "a/c.csv.gz", "a/b/c.csv", "a/b/c.csv.gz", "a/c/c.csv", "a/c/c.csv.gz", "a/b/c/d.csv", "a/b/c/d.csv.gz"}, set()),
+    (["*.csv", "*.gz"], {"a.csv", "a.csv.gz"}, set()),
+    (["a/*.csv", "a/*/*.csv"], {"a/b.csv", "a/c.csv", "a/b/c.csv", "a/c/c.csv"}, {"a"}),
+    (["a/*.csv", "a/b/*.csv"], {"a/b.csv", "a/c.csv", "a/b/c.csv"}, {"a", "a/b"}),
+]
+
+
+@pytest.mark.parametrize(
+    "globs,expected_matches,expected_path_prefixes",
+    cases,
```

Comment:
nit: is the plan to reuse these cases across more test methods? If we don't plan to reuse these, it might make more sense to just declare each one as `pytest.param([], set(FILEPATHS), set(), id="test_name_whatever")` and define them within the `parametrize` block like we do other tests. I'm not sure the id loop after makes it easier to read